### DISCUSSION TITLE
分享一下我的clash verge的Tun模式的全局扩展配置：防dns泄露，换geosite和ip文件，fakeip，过滤自己的域名，与蒲公英组网兼容

### DISCUSSION THREAD

**Discussion by SyberRabbit** [clash-verge-rev/clash-verge-rev#2368] [Show and tell]
# 我也是个刚入门槛的萌新，有如题需求的萌新，欢迎来回帖，大佬也可以给我一些配置建议，谢谢。
## 版本：
Verge版本：v2.0.2
Clash内核：v1.18.10 Mihomo
#### 如果有组网需求：
需要配置的是最后的extra-config
在cmd中输入ipconfig以及route print，来分析一下你的组网的虚拟网卡名称是什么，我这里蒲公英组网的虚拟网卡叫以太网 2
ip: "192.168.xxx.0/24"这个配置的话，看看你有没有设置路由器的lan口和dhcp为192.168.xxx.0这一段的ip，一般默认的ip段是192.168.1.0/24，`不过我是后来在外面连接其他wifi后，发现组网后无法访问家里的设备，看了route发现，路由到了外面wifi的路由器上了，而不是路由到蒲公英的虚拟网卡，我这才知道这是ip段冲突了，因为外面路由器的wifi的ip段也是192.168.1.0，于是就把xxx这一部分改了。`
##### 如果你不光有组网需求，而且还把内网的私有ip，进行ddns了，那还需要配置fake-ip-filter：
我把*.g470.example.xyz指定到192.168.xxx.106了，如果只设置了extra-config，那么域名会解析成fake-ip指定的虚假ip段，然后这个ip传到蒲公英虚拟网卡，那肯定是找不到对应主机的。`这里我解释的总感觉怪怪的，我可能说错了，请大佬纠正。`所以我干脆用+.example.xyz将我所有子域名和子子子域名都正常解析好了
#### rules中的MATCH,节点选择
这个“节点选择”也是需要改的，直接看图就能明白，图在文章最后
## 全局扩展配置在这儿
```# Profile Enhancement Merge Template for Clash Verge

profile:
  store-selected: true

geodata-mode: false
geox-url:
  geosite: "https://cdn.jsdelivr.net/gh/MetaCubeX/meta-rules-dat@release/geosite-lite.dat"
  geoip: "https://cdn.jsdelivr.net/gh/MetaCubeX/meta-rules-dat@release/geoip-lite.dat"
  mmdb: "https://cdn.jsdelivr.net/gh/MetaCubeX/meta-rules-dat@release/country-lite.mmdb"
geo-auto-update: true
geo-update-interval: 24
dns:
  use-system-hosts: false
  enable: true
  fake-ip-filter:
    - '+.example.xyz'
  nameserver:
    - https://1.1.1.1/dns-query
    - https://8.8.8.8/dns-query
    - 1.1.1.1
    - 8.8.8.8
  nameserver-policy:
    geosite:cn:
        - https://1.12.12.12/dns-query
        - https://223.5.5.5/dns-query
  proxy-server-nameserver:
    - https://1.12.12.12/dns-query
    - https://223.5.5.5/dns-query
rules:
  - DOMAIN-SUFFIX,googleadservices.com,DIRECT
  - DOMAIN-SUFFIX,imgse.com,DIRECT
  - DOMAIN-SUFFIX,s21.ax1x.com,DIRECT
  - DOMAIN-SUFFIX,example.xyz,DIRECT
  # 兜底路由
  - GEOSITE,CN,DIRECT
  - GEOIP,CN,DIRECT
  - MATCH,节点选择
extra-config:
  - name: "route-192.168.xxx.0"
    type: "static"
    static-routes:
      - ip: "192.168.xxx.0/24"
        via: "以太网 2"
```
![图片](https://github.com/user-attachments/assets/c085b376-1505-45ce-84b2-b6c53f479f0c)

**Comment by soberlee97** [comment]
proxy-server-nameserver 为啥还填国内的，你这不还是泄露了吗

**Reply by SyberRabbit** [reply]
对，这个不好，泄漏的是节点域名，当时没找到免费公益可稳定直连的doh，也没有稳定的单独节点供代理doh，是无奈之举
现在我已经有了公益doh，全无脑一个doh就行了

**Reply by SyberRabbit** [reply]
> proxy-server-nameserver 为啥还填国内的，你这不还是泄露了吗

这个是我现在用的
https://raw.githubusercontent.com/SyberRabbit/About-Clash/refs/heads/clash-all-client-yaml/config.yaml
